### DISCUSSION TITLE
Error when use RAND_bytes with openssl version 3.1.1

### DISCUSSION THREAD

**Discussion by nickychen123456** [openssl/openssl#22850] [Q&A]
I used the openssl 3.1.1 in my app on windows system, when I use RAND_bytes api I got the error "ssl_ctx_t::export_ssl_error , ssl_errno = 301990032, error msg = error:12000090:random number generator::unable to fetch drbg", what is the root cause for this problem, how can I resolve this?

**Comment by nhorman** [comment]
Can you provide a reproducer?  That error would be produced if rand_new_drbg was unable to find the requested rng, which really shouldn't happen unless you are requesting a rng that doesn't have an underlying provider

it would also be good to know if running openssl rand on your platform works properly

**Comment by mattcaswell** [comment]
> I used the openssl 3.1.1 in my app on windows system, when I use RAND_bytes api I got the error "ssl_ctx_t::export_ssl_error , ssl_errno = 301990032, error msg = error:12000090:random number generator::unable to fetch drbg", what is the root cause for this problem, how can I resolve this?

How do you create your `SSL_CTX`? i.e. do you create it with `SSL_CTX_new` or `SSL_CTX_new_ex`? In other words are you passing an explicit libctx to the `SSL_CTX` or are you using the default libctx? Does the libctx you are using have the default provider loaded into it (which should happen automatically unless you have explicitly loaded other providers).
